src/commands/ai/agents/call.ts

```diff
@@ -0,0 +1,234 @@
+import {ReadableStream} from 'node:stream/web'
+
+type CLIParseError = CLIParseErrorOptions & {
+  parse: {
+    input: string,
+    output: ParserOutput<Call>
+  }
+}
+
+export type AgentRequest = {
+  model: string;
+  messages: Array<{
+    role: string;
+    content: string;
+  }>;
+  max_tokens_per_inference_request?: number;
+  stop?: string[];
+  temperature?: number;
+  tools?: Array<{
+    type: string;
+    name: string;
+    description?: string;
+    runtime_params?: {
+      target_app_name?: string;
+      tool_params?: {
+        cmd?: string;
+        description?: string;
+        parameters?: {
+          type: string;
+          properties: Record<string, unknown>;
+          required: string[];
+        };
+      };
+    };
+  }>;
+  top_p?: number;
+}
+
+export default class Call extends Command {
+  static args = {
+    model_resource: Args.string({
+      description: 'resource ID or alias of model (--app flag required if alias is used)',
+      required: false,
+      default: 'heroku-inference',
+    }),
+  }
+
+  static description = 'make an inference request to the Heroku Inference Agents API'
+  static examples = [
+    'heroku ai:agents:call my_llm --app my-app --prompt "What is the current time?"',
+    'heroku ai:agents:call my_llm --app my-app --messages \'[{"role":"user","content":"What is the current time?"}]\'',
+  ]
+
+  static flags = {
+    app: flags.app({
+      required: false,
+      description: 'name or ID of app (required if alias is used)',
+    }),
+    json: flags.boolean({char: 'j', description: 'output response as JSON'}),
+    optfile: flags.string({
+      description: 'additional options for model inference, provided as a JSON config file',
+      required: false,
+    }),
+    opts: flags.string({
+      description: 'additional options for model inference, provided as a JSON string',
+      required: false,
+    }),
+    output: flags.string({
+      char: 'o',
+      description: 'file path where command writes the model response',
+      required: false,
+    }),
+    prompt: flags.string({
+      char: 'p',
+      description: 'input prompt for model (will be converted to a user message)',
+      required: false,
+    }),
+    messages: flags.string({
+      description: 'JSON array of messages to send to the model',
+      required: false,
+    }),
+    remote: flags.remote(),
+  }
+
+  public async run(): Promise<void> {
+    let flags = {} as ParserOutput<Call>['flags']
+    let args = {} as ParserOutput<Call>['args']
+    try {
+      ({args, flags} = await this.parse(Call))
+    } catch (error) {
+      const {parse: {output}} = error as CLIParseError
+      ({args, flags} = output)
+    }
+
+    const {model_resource: modelResource} = args
+    const {app, json, optfile, opts, output, prompt, messages} = flags
+
+    if (!prompt && !messages && !optfile && !opts) {
+      throw new Error('You must provide either --prompt, --messages, --optfile, or --opts.')
+    }
```

Comment:
You can use oclif's `exactlyOne` flag property for this. [Example](https://github.com/heroku/cli/blob/a8a0e0c3df73abe59b614410b90be255b3abbbd0/packages/cli/src/commands/ci/config/get.ts#L19). [Reference](https://oclif.io/docs/flags/).